File: Syncfusion_Sample/Windows/Tools.Windows/Samples/2.0/Office Style Controls/Office Style Form/VB/MainForm.vb
#Region "Copyright Syncfusion Inc. 2001 - 2013"
' Copyright Syncfusion Inc. 2001 - 2013. All rights reserved.
' Use of this code is subject to the terms of our license.
' A copy of the current license can be obtained at any time by e-mailing
' licensing@syncfusion.com. Any infringement will be prosecuted under
' applicable laws. 
#End Region

Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports Syncfusion.Windows.Forms

Namespace Office2007Form_2005
	Public Partial Class MainForm
		Inherits Office2007Form
		#Region "Constructor"
		Public Sub New()
			InitializeComponent()
			Me.checkBoxAdv3.Checked = True

			For Each str As String In System.Enum.GetNames(GetType(FontStyle))
				Me.comboBox3.Items.Add(str)
			Next str

			Me.comboBox3.SelectedIndex = 1
            Me.comboBox1.SelectedIndex = 0
            Me.ColorPickerButton1.SelectedColor = Color.Green
            Me.CaptionForeColor = Me.ColorPickerButton1.SelectedColor
            Me.ColorPickerButton1.SelectedAsText = True
		End Sub
		#End Region

		#Region "Themes"
		Private Sub radioButtonAdv1_CheckChanged(ByVal sender As Object, ByVal e As EventArgs) Handles radioButtonAdv1.CheckChanged
			Me.ColorScheme = Office2007Theme.Blue
			Me.BackColor = Color.FromArgb(241, 244, 254)

		End Sub

		Private Sub radioButtonAdv2_CheckChanged(ByVal sender As Object, ByVal e As EventArgs) Handles radioButtonAdv2.CheckChanged
			Me.ColorScheme = Office2007Theme.Silver
			Me.BackColor = Color.FromArgb(202, 207, 217)
		End Sub

		Private Sub radioButtonAdv3_CheckChanged(ByVal sender As Object, ByVal e As EventArgs) Handles radioButtonAdv3.CheckChanged
			Me.ColorScheme = Office2007Theme.Black
			Me.BackColor = Color.FromArgb(113, 113, 113)
		End Sub
		#End Region

		#Region "Help Button"
		Private Sub checkBoxAdv1_CheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkBoxAdv1.CheckStateChanged
			If Me.checkBoxAdv1.Checked Then
				Me.MaximizeBox = False
				Me.MinimizeBox = False
			Else
				Me.MaximizeBox = False
				Me.MinimizeBox = True
			End If

		End Sub
		#End Region 

		#Region "RightToLeft"

		Private Sub checkBoxAdv2_CheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkBoxAdv2.CheckStateChanged
			If Me.checkBoxAdv2.Checked Then
				Me.RightToLeft = RightToLeft.Yes
			Else
				Me.RightToLeft = RightToLeft.No
			End If
		End Sub
		#End Region

		#Region "Show Icon"
		Private Sub checkBoxAdv3_CheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkBoxAdv3.CheckStateChanged
			If Me.checkBoxAdv3.Checked Then
				Me.ShowIcon = True
			Else
				Me.ShowIcon = False
			End If
		End Sub
		#End Region

		#Region "Caption settings"
		Private Sub comboBox1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBox1.SelectedIndexChanged
			Select Case comboBox1.SelectedIndex
				Case 0
					Me.CaptionAlign = HorizontalAlignment.Left
				Case 1
					Me.CaptionAlign = HorizontalAlignment.Center
				Case 2
					Me.CaptionAlign = HorizontalAlignment.Right
			End Select
		End Sub

		Private Sub comboBox2_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBox2.SelectedIndexChanged
			SetCaptionFont()
		End Sub

        Protected Sub SetCaptionFont()
            If Not (comboBox3.SelectedItem Is Nothing) Then
                Dim fs As FontStyle = CType(System.Enum.Parse(GetType(FontStyle), Me.comboBox3.SelectedItem.ToString(), True), FontStyle)
                Me.CaptionFont = New Font(Me.comboBox2.SelectedItem.ToString(), CInt(Fix(Me.numericUpDown1.Value)), fs)
            End If
        End Sub

		Private Sub numericUpDown1_ValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles numericUpDown1.ValueChanged
			SetCaptionFont()
		End Sub

		Private Sub comboBox3_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBox3.SelectedIndexChanged
			SetCaptionFont()
        End Sub

        Private Sub ColorPickerButton1_ColorSelected(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ColorPickerButton1.ColorSelected
            Me.CaptionForeColor = Me.ColorPickerButton1.SelectedColor
        End Sub
#End Region
        
    End Class
End Namespace

File: Syncfusion_Sample/Windows/Tools.Windows/Samples/2.0/Office Style Controls/Office Style Form/CS/MainForm.cs
#region Copyright Syncfusion Inc. 2001 - 2013
// Copyright Syncfusion Inc. 2001 - 2013. All rights reserved.
// Use of this code is subject to the terms of our license.
// A copy of the current license can be obtained at any time by e-mailing
// licensing@syncfusion.com. Any infringement will be prosecuted under
// applicable laws. 
#endregion
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Syncfusion.Windows.Forms;

namespace Office2007Form_2005
{
    public partial class MainForm : Office2007Form
    {
        #region Constructor
        public MainForm()
        {
            InitializeComponent();
            this.checkBoxAdv3.Checked = true;
            
            foreach (String str in Enum.GetNames(typeof(FontStyle)))
                this.comboBox3.Items.Add(str);

            this.comboBox3.SelectedIndex = 1;
            this.comboBox1.SelectedIndex = 0;
            this.colorPickerButton1.SelectedColor = Color.Green;
            this.CaptionForeColor = this.colorPickerButton1.SelectedColor;
            this.colorPickerButton1.SelectedAsText = true;
        }
        #endregion

        #region Themes
        private void radioButtonAdv1_CheckChanged(object sender, EventArgs e)
        {
            this.ColorScheme = Office2007Theme.Blue;
            this.BackColor = Color.FromArgb(241, 244, 254);          
            
        }

        private void radioButtonAdv2_CheckChanged(object sender, EventArgs e)
        {
            this.ColorScheme = Office2007Theme.Silver;
            this.BackColor = Color.FromArgb(202, 207, 217);
       }

        private void radioButtonAdv3_CheckChanged(object sender, EventArgs e)
        {
            this.ColorScheme = Office2007Theme.Black;
            this.BackColor = Color.FromArgb(113, 113, 113);
             }
        #endregion

        #region Help Button
        private void checkBoxAdv1_CheckStateChanged(object sender, EventArgs e)
        {
            if (this.checkBoxAdv1.Checked)
            {
                this.MaximizeBox = false;
                this.MinimizeBox = false;
            }
            else
            {
                this.MaximizeBox = false;
                
                this.MinimizeBox = true;
            }

        }
        #endregion 

        #region RightToLeft

        private void checkBoxAdv2_CheckStateChanged(object sender, EventArgs e)
        {
            if (this.checkBoxAdv2.Checked)
            {
                this.RightToLeft = RightToLeft.Yes;
            }
            else
            {
                this.RightToLeft = RightToLeft.No;
            }
        }
        #endregion

        #region Show Icon
        private void checkBoxAdv3_CheckStateChanged(object sender, EventArgs e)
        {
            if (this.checkBoxAdv3.Checked)
            {
                this.ShowIcon = true;
            }
            else
            {
                this.ShowIcon = false;
            }
        }
        #endregion

        #region Caption settings
        private void comboBox1_SelectedIndexChanged(object sender, EventArgs e)
        {
            switch (comboBox1.SelectedIndex)
            {
                case 0:
                    this.CaptionAlign = HorizontalAlignment.Left;
                    break;
                case 1:
                    this.CaptionAlign = HorizontalAlignment.Center;
                    break;
                case 2:
                    this.CaptionAlign = HorizontalAlignment.Right;
                    break;
            }
        }

        private void comboBox2_SelectedIndexChanged(object sender, EventArgs e)
        {
            SetCaptionFont();
        }

        protected void SetCaptionFont()
        {
            FontStyle fs = (FontStyle)Enum.Parse(typeof(FontStyle), this.comboBox3.SelectedItem.ToString(), true);
            this.CaptionFont = new Font(this.comboBox2.SelectedItem.ToString(), (int)this.numericUpDown1.Value, fs);
        
        }

        private void numericUpDown1_ValueChanged(object sender, EventArgs e)
        {
            SetCaptionFont();
        }

        private void comboBox3_SelectedIndexChanged(object sender, EventArgs e)
        {
            SetCaptionFont();
        }

        private void colorPickerButton1_ColorSelected(object sender, EventArgs e)
        {
            this.CaptionForeColor = this.colorPickerButton1.SelectedColor;
        }
        #endregion
        
    }
}